File: smallpict-java/src/main/java/com/smallpict/errors/NotFoundException.java
package com.smallpict.errors;

import java.util.Map;

public class NotFoundException extends SmallPictException {
    public NotFoundException(String message) {
        this(message, null, null);
    }

    public NotFoundException(String message, String requestId, Map<String, Object> details) {
        super(message, "NOT_FOUND", 404, requestId, details, null);
    }
}

File: smallpict-php/src/Exceptions/NotFoundException.php
<?php

declare(strict_types=1);

namespace SmallPict\Exceptions;

class NotFoundException extends SmallPictException
{
    /**
     * @param string $message
     * @param string|null $requestId
     * @param array<string, mixed> $details
     * @param \Throwable|null $previous
     */
    public function __construct(
        string $message = 'Resource or job ID not found',
        ?string $requestId = null,
        array $details = [],
        ?\Throwable $previous = null
    ) {
        parent::__construct($message, 'NOT_FOUND', 404, $requestId, $details, $previous);
    }
}
